execution/gethexec/executionengine.go

```diff
@@ -661,6 +662,10 @@ func (s *ExecutionEngine) createBlockFromNextMessage(msg *arbostypes.MessageWith
 	statedb.StartPrefetcher("TransactionStreamer")
 	defer statedb.StopPrefetcher()
 
+	runMode := core.MessageCommitMode
+	if isMsgForPrefetch {
+		runMode = core.MessageReplayMode
```

Comment:
good point, Replay is not a best name here. 

The MessageReplayMode is meant for execution other then adding to the chain head, so that is not only for prefetch. It seems that we need a name that will cover both replayed msgs and those run for the first time for old state and with results not committed to the db. 

does `MessageEphemeralMode` sound good?

Reply:
I don't love `MessageEphemeralMode` because it's too general, as eth_call and eth_estimateGas are ephemeral too.

Reply:
just an idea, not sure if that's worth the added complexity, but it maybe could make it cleaner (?) - how about defining the modes as struct: 

```
struct MessageRunMode {
    Ephemeral bool
    Mutating bool
    ExecutedOnChain bool
    GasEstimating bool
    ....
}

var MessageCommitMode = MessageRunMode {
    Ephemeral: false,
    Mutating: true,
    ExecutedOnChain: true, 
} 

var MessageEthcallMode = MessageRunMode {
    Ephemeral: true,
    Mutating: false,
    ExecutedOnChain: false,
}

```

Reply:
or we could just make those fields bit flags and the modes would be eg. `MessageCommitMode = Mutating | ExecutedOnChain `

Reply:
or we could just add `MessagePrefetchMode` like that:
```
const (
    MessageCommitMode MessageRunMode = iota
    ...
    MessageReplayMode
    MessagePrefetchMode = MessageReplayMode
)
```

but I am not sure if that's not more convoluted

Reply:
I think we'd want to replace MessageCommitMode with a struct.
This struct will also have things like wasm-targets, which are currently passed uncleanly via ethdb.
But that's a separate PR.

Reply:
for now I am leaving `MessageReplayMode` here, added a follow-up ticket for the MessageRunMode refactor: NIT-2827